### PR TITLE
Feat/headers

### PR DESCRIPTION
**Pull request by titoine54**
Added support for custom headers.

Fixes issue #3 and #4

### CODE REVIEW

File: __tests__/integration.test.ts [hunk 1]
```diff
@@ -77,6 +79,30 @@ describe('when called with a GraphQL query', () => {
   })
 })
 
+describe('when called with a custom header containing wrong auth', () => {
+  beforeEach(() => {
+    process.env['INPUT_URL'] = 'https://api.github.com'
+    process.env['INPUT_HEADERS'] =
+      '{"content-type":"application/json", "Authorization": "Basic YWxhZGRpbjpvcGVuc2VzYW1l"}'
```
Comment (CamiloGarciaLaRotta) [inline __tests__/integration.test.ts:86]: Some people actually try to scrape tokens out of GH. Let's just be clear that this is a dummy value.
I'm cool with whatever name you want 🙌🏽
```suggestion
      '{"content-type":"application/json", "Authorization": "Basic pho-bac"}'
```

File: README.md [hunk 2]
```diff
@@ -11,6 +11,7 @@ A Typescript Action that performs HTTP requests within your workflow. It support
 | url | Endpoint to query | https://api.github.com/graphql |
 | method | HTTP method | `GET` |
 | data | HTTP request payload |  |
+| headers | HTTP headers | {'Content-Type': 'application/json'} |
```
Comment (CamiloGarciaLaRotta) [inline README.md:14]: This is technically not true for all inputs 😅the default values are defined by by the [`actions.yml`](https://github.com/CamiloGarciaLaRotta/watermelon-http-client/pull/25/files#diff-689e1a63f106e616debbed3b5aedcb43R15-R16). But you are correct with respect to GraphQL.
```suggestion
| headers | HTTP headers | Empty by default. If no headers are supplied, GraphQL queries will be sent with `{'Content-Type': 'application/json'}` |
```

File: src/http.ts [hunk 3]
```diff
@@ -1,39 +1,41 @@
...
       url,
       method,
       data,
-      headers: {'Content-Type': 'application/json'}
+      headers
     })
 
     const status = response.status
-    const headers = response.headers
-    const payload = response.data as Object
```
Comment (CamiloGarciaLaRotta) [inline src/http.ts:26]: 💪🏽 noiice

File: README.md [hunk 4]
```diff
@@ -31,6 +32,7 @@ A Typescript Action that performs HTTP requests within your workflow. It support
   uses: CamiloGarciaLaRotta/watermelon-http-client@v1
   with:
     url: 'https://countries.trevorblades.com/'
+    headers: '{"Content-Type": "application/json"}'
```
Comment (CamiloGarciaLaRotta) [inline README.md:35]: 🤔For this examples section I was thinking of giving the minimal copy-pasteable snippets that work out-of the box. In which case I think we can remove this line:

```suggestion
```

One thing that is nice so far is that we have "sane defaults", meaning that sending no custom headers, defaults to the appropriate GraphQL headers. You can see that it succeed in the [GraphQL workflow](https://github.com/CamiloGarciaLaRotta/watermelon-http-client/runs/562788947#step:3:21) without us needing to defining the headers:
https://github.com/CamiloGarciaLaRotta/watermelon-http-client/blob/f2870ebecac7918a018b83fa86c7e04ec3452047/.github/workflows/graphql.yml#L17-L25

File: README.md [hunk 5]
```diff
@@ -43,6 +45,7 @@ A Typescript Action that performs HTTP requests within your workflow. It support
 - [`GET`](.github/workflows/get.yml)
   ```yaml
   uses: CamiloGarciaLaRotta/watermelon-http-client@v1
+  headers: '{"Authorization": "Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ=="}'
```
Comment (CamiloGarciaLaRotta) [inline README.md:48]: bis: same point regarding headers 🙂
```suggestion
```

File: README.md [hunk 6]
```diff
@@ -52,6 +55,7 @@ A Typescript Action that performs HTTP requests within your workflow. It support
   uses: CamiloGarciaLaRotta/watermelon-http-client@v1
   with:
     url: 'https://jsonplaceholder.typicode.com/todos'
+    headers: '{ "Connection": "keep-alive"}'
```
Comment (CamiloGarciaLaRotta) [inline README.md:58]: bis: same point regarding headers
```suggestion
```

File: README.md (hunk 4 above)
Comment (CamiloGarciaLaRotta) [reply README.md:35]: On the other hand, I see the point of giving an example of how to define custom headers. When we merge https://github.com/CamiloGarciaLaRotta/watermelon-http-client/pull/26 we can refer to that file as an example directly 👍🏽

File: __tests__/integration.test.ts (hunk 1 above)
Comment (titoine54) [reply __tests__/integration.test.ts:86]: changed it to : DummyToken123

File: README.md (hunk 2 above)
Comment (titoine54) [reply README.md:14]: Changed it to : DummyToken123

File: README.md (hunk 6 above)
Comment (titoine54) [reply README.md:58]: removed

File: README.md (hunk 5 above)
Comment (titoine54) [reply README.md:48]: removed

File: README.md (hunk 4 above)
Comment (titoine54) [reply README.md:35]: I agree. Removed !

File: README.md (hunk 2 above)
Comment (titoine54) [reply README.md:14]: ooh that's true. My bad. Fixed.

**Pull request merged**